Source/Core/DolphinQt/MainWindow.cpp

```diff
@@ -1,362 +1,302 @@
-		tr("All supported ROMs (%1);;All files (*)")
-		.arg(SL("*.gcm *.iso *.ciso *.gcz *.wbfs *.elf *.dol *.dff *.tmd *.wad")));
+	m_stack = new QStackedWidget;
+	m_stack->setMinimumSize(640, 480);
+	m_stack->addWidget(m_game_table);
+	m_stack->addWidget(m_render_widget);
+	setCentralWidget(m_stack);
 }
 
-QString DMainWindow::ShowFolderDialog()
+void MainWindow::PopulateToolBar()
 {
-	return QFileDialog::getExistingDirectory(this, tr("Browse for a directory to add"),
-	                                         QDir::homePath(),
-	                                         QFileDialog::ShowDirsOnly);
+	m_tool_bar->clear();
+
+	QString dir = QString::fromStdString(File::GetThemeDir(SConfig::GetInstance().theme_name));
+	m_tool_bar->addAction(QIcon(dir + tr("open.png")), tr("Open"), this, SLOT(Open()));
+	m_tool_bar->addAction(QIcon(dir + tr("refresh.png")), tr("Refresh"), this, SLOT(Refresh()));
+	m_tool_bar->addAction(QIcon(dir + tr("browse.png")), tr("Browse"), this, SLOT(Browse()));
+
+	m_tool_bar->addSeparator();
+
+	QAction* play_action = m_tool_bar->addAction(QIcon(dir + tr("play.png")), tr("Play"), this, SLOT(Play()));
+	connect(this, &MainWindow::EmulationStarted, [=](){ play_action->setEnabled(false); });
+	connect(this, &MainWindow::EmulationPaused, [=](){ play_action->setEnabled(true); });
+	connect(this, &MainWindow::EmulationStopped, [=](){ play_action->setEnabled(true); });
+
+	QAction* pause_action = m_tool_bar->addAction(QIcon(dir + tr("pause.png")), tr("Pause"), this, SLOT(Pause()));
+	pause_action->setEnabled(false);
+	connect(this, &MainWindow::EmulationStarted, [=](){ pause_action->setEnabled(true); });
+	connect(this, &MainWindow::EmulationPaused, [=](){ pause_action->setEnabled(false); });
+	connect(this, &MainWindow::EmulationStopped, [=](){ pause_action->setEnabled(false); });
+
+	QAction* stop_action = m_tool_bar->addAction(QIcon(dir + tr("stop.png")), tr("Stop"), this, SLOT(Stop()));
+	stop_action->setEnabled(false);
+	connect(this, &MainWindow::EmulationStarted, [=](){ stop_action->setEnabled(true); });
+	connect(this, &MainWindow::EmulationPaused, [=](){ stop_action->setEnabled(true); });
+	connect(this, &MainWindow::EmulationStopped, [=](){ stop_action->setEnabled(false); });
+
+	m_tool_bar->addAction(QIcon(dir + tr("fullscreen.png")), tr("Full Screen"))->setEnabled(false);
+	m_tool_bar->addAction(QIcon(dir + tr("screenshot.png")), tr("Screen Shot"))->setEnabled(false);
+	m_tool_bar->addSeparator();
+	m_tool_bar->addAction(QIcon(dir + tr("config.png")), tr("Settings"))->setEnabled(false);
+	m_tool_bar->addAction(QIcon(dir + tr("graphics.png")), tr("Graphics"))->setEnabled(false);
+	m_tool_bar->addAction(QIcon(dir + tr("classic.png")), tr("Controllers"))->setEnabled(false);
 }
 
-void DMainWindow::DoStartPause()
+void MainWindow::PopulateGameTable()
 {
-	if (Core::GetState() == Core::CORE_RUN)
+	m_game_table->setEnabled(false);
+	while (m_game_table->rowCount() > 0)
+		m_game_table->removeRow(0);
+	m_games.clear();
```

Comment:
This bit of code (from the beginning of the function to here) should be extracted into its own function (ClearGameTable or some better name)